tests/integration/security/reachability_test.go

```diff
@@ -112,21 +120,36 @@ func TestReachability(t *testing.T) {
 					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
 						// autoMtls doesn't work for client that doesn't have proxy, unless target doesn't
 						// have proxy or have mTLS disabled
-						if src == rctx.Naked {
-							return opts.Target == rctx.Naked || (opts.Target == rctx.B && opts.PortName != "http")
+						if rctx.IsNaked(src) {
+							return rctx.IsNaked(opts.Target) || (opts.Target == rctx.B && opts.PortName != "http")
 
 						}
+						// headless with sidecar injected, global mTLS enabled, no client side transport socket or transport_socket_matches since it's headless service.
+						if src != rctx.Headless && opts.Target == rctx.Headless {
+							return false
+						}
 						// PeerAuthentication disable mTLS for workload app:b, except http port. Thus, autoMTLS
 						// will fail on all ports on b, except http port.
 						return opts.Target != rctx.B || opts.PortName == "http"
 					},
 				},
+				{
+					// test access headless service with mTLS off to simulate without sidecar.
+					ConfigFile: "beta-mtls-off-headless.yaml",
```

Comment:
why is this not covered by `beta-mtls-off`?

Reply:
Right, it is covered